ui/proposals_page.go

```diff
@@ -541,24 +496,45 @@ func (pg *proposalsPage) layoutStartSyncSection(gtx C) D {
 	})
 }
 
-func (pg *proposalsPage) layoutSyncSection(gtx C, common pageCommon) D {
+func (pg *proposalsPage) layoutSyncSection(gtx C) D {
 	if pg.isSynced {
-		common.refreshPage()
 		return pg.layoutIsSyncedSection(gtx)
 	} else if pg.wallet.IsSyncingProposals() {
 		return pg.layoutIsSyncingSection(gtx)
 	}
 	return pg.layoutStartSyncSection(gtx)
 }
 
-func (pg *proposalsPage) Layout(gtx C, common pageCommon) D {
-	if !pg.hasFetchedInitialProposals && !pg.isFetchingInitialProposals {
-		pg.fetchProposals()
+func (pg *proposalsPage) initializeProposaltabItems() {
+	pg.proposalsItemSet = true
+	if len((*pg.proposals).Proposals) == 0 {
+		pg.wallet.SyncProposals()
+		pg.proposalsItemSet = false
 	}
 
-	if !pg.hasRegisteredListeners {
-		pg.wallet.AddProposalNotificationListener(proposalNotificationListeners{pg})
-		pg.hasRegisteredListeners = true
+	for i := range (*pg.proposals).Proposals {
+		if i != len((*pg.proposals).Proposals) {
+			item := proposalItem{
+				btn:               new(widget.Clickable),
+				proposal:          (*pg.proposals).Proposals[i],
+				voteBar:           pg.theme.VoteBar(pg.infoIcon, pg.legendIcon),
+				infoIcon:          pg.infoIcon,
+				stateInfoTooltip:  pg.theme.Tooltip(),
+				stateTooltipLabel: pg.theme.Caption(""),
+			}
+			for k := range pg.tabs.tabs {
+				if pg.tabs.tabs[k].category == (*pg.proposals).Proposals[i].Category {
+					pg.tabs.tabs[k].proposals = append(pg.tabs.tabs[k].proposals, item)
+					break
+				}
+			}
+		}
+	}
+}
```

Comment:
This looks like a duplicate of the code in `addDiscoveredProposal` and the method can simply be called here.